tests/js/app/widgets/testFormattableLabel.js

```diff
@@ -55,4 +56,29 @@ describe('Formattable Label', function () {
         label.label = "Custer&apos;s Last Stand";
         expect(label.label).toBe("CUSTER&apos;S LAST STAND");
     });
+
+    it('handles ampersand symbols correctly', function () {
+        let cssProvider = new Gtk.CssProvider();
+
+        cssProvider.load_from_data ("* { \
+            -EknFormattableLabel-text-transform: 'uppercase';\
+        }");
+
+        let win = new Gtk.OffscreenWindow();
+        win.add(label);
+        win.show_all();
+        label.get_style_context().add_provider(cssProvider,
+                                                Gtk.StyleProvider.PRIORITY_APPLICATION + 10);
+
+        Utils.update_gui();
+        let text = "a & b";
+
+        label.use_markup = true;
+        label.label = GLib.markup_escape_text(text, -1);
```

Comment:
Why do you need to call `GLib.markup_escape_text(text, -1);` here? Won't the fact that you have `use_markup = true` plus that custom label setter that calls  `_format_label` mean this gets called already with your changes above?